src/System.Private.CoreLib/shared/System/Attribute.cs

```diff
@@ -74,15 +75,15 @@ public override int GetHashCode()
                 if (vThis != null)
                     return vThis.GetHashCode();
 
-                type = type.BaseType;
+                type = type.BaseType!;
```

Comment:
where is the `!` coming from? Is BaseType nullable? If so can this ever NRE?

Reply:
> where is the ! coming from? Is BaseType nullable? 

Type.BaseType could return null, as you know if any of the overrides return null base will be nullable, that applies here

> If so can this ever NRE?

I would say could be, but not sure if i should add additional check

Reply:
BaseType can always be object in which case we stop the loop